### PR TITLE
Password reset tracking

### PR DESCRIPTION
**Pull request by inaod568**
New table for keeping track of the password resets. Password reset links should only be valid once.

We can also later use build on this to make sure that after the user has changed their password, their current tokens are no longer valid. This can be checked by comparing the `issued` with the token creation date and checking if the password reset link is `valid` (if it has been used or not).

- [x] Tests passing
- [x] Black formatting
- [x] Migrations for any changes to the database schema
- [x] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -49,3 +49,4 @@ Please add a _short_ line describing the PR you make, if the PR implements a spe
 - Display an error message when the user makes too many authentication requests. ([#1034](https://github.com/ScilifelabDataCentre/dds_web/pull/1034))
 - New endpoint for Unit Personnel and Admins to list the other Unit Personnel / Admins within their project ([#1050](https://github.com/ScilifelabDataCentre/dds_web/pull/1050))
 - Make previous HOTP invalid at password reset ([#1054](https://github.com/ScilifelabDataCentre/dds_web/pull/1054))
+- New PasswordReset table to keep track of when a user has requested a password reset ()
```
Comment (inaod568) [inline CHANGELOG.md:52]:
```suggestion
- New PasswordReset table to keep track of when a user has requested a password reset ([#1058](https://github.com/ScilifelabDataCentre/dds_web/pull/1058))
```

File: dds_web/database/models.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 # Standard library
 import datetime
+from enum import unique
```
Comment (inaod568) [inline dds_web/database/models.py:9]:
```suggestion
```

File: dds_web/web/user.py [hunk 3]
```diff
@@ -336,6 +336,20 @@ def request_reset_password():
                 ),
                 additional_claims={"rst": "pwd"},
             )
+
+            # Create row in password reset table
+            ongoing_password_reset = models.PasswordReset.query.filter_by(
+                email=email.email
+            ).one_or_none()
+            if ongoing_password_reset:
+                ongoing_password_reset.issued = dds_web.utils.current_time()
```
Comment (inaod568) [inline dds_web/web/user.py:345]:
```suggestion
                ongoing_password_reset.issued = dds_web.utils.current_time()
                ongoing_password_reset.valid = True
```

**Review by zishanmirza — APPROVED** [review]

**Review by aanil — APPROVED** [review]
Functionality seems to work!

**Pull request merged**